components/VisualEditorInput/src/VisualEditorInput.tsx

```diff
@@ -18,177 +16,49 @@ const Container = styled.div<{ size: 'medium' | 'small' }>`
-    border-style: solid;
-    border-color: ${themeVars.layout.strokes};
-
-    background: ${({ variation }) => (variation === 'filled' ? themeVars.layout.surfaceLow : 'transparent')};
-    &[data-has-error='true'] {
-        border-color: ${themeVars.actions.error};
-        outline-width: ${themeVars.stroke.thin};
-        outline-style: solid;
-        outline-color: ${themeVars.actions.error};
-    }
-
-    &:hover {
-        border-color: ${themeVars.layout.onSurface.tertiary};
-    }
-
-    &:active {
-        border-color: ${themeVars.layout.onSurface.tertiary};
-        outline-width: ${themeVars.stroke.thin};
-        outline-style: solid;
-        outline-color: ${themeVars.layout.onSurface.tertiary};
-    }
-
-    &:focus-within {
-        border-color: ${themeVars.actions.primary};
-        outline-width: ${themeVars.stroke.thin};
-        outline-style: solid;
-        outline-color: ${themeVars.actions.primary};
-    }
-
-    &[data-disabled='true'] {
-        cursor: not-allowed;
-    }
-
-    transition:
-        border-color ${themeVars.animation.duration.fast} ${themeVars.animation.timing.easeOut},
-        outline-color ${themeVars.animation.duration.fast} ${themeVars.animation.timing.easeOut};
-`;
-
-const CustomInput = styled.input`
-    box-sizing: border-box;
-    width: 100%;
-
-    font-family: Inter;
-    font-size: 14px;
-    font-style: normal;
-    font-weight: 400;
-    line-height: 20px;
-
-    color: ${themeVars.layout.onSurface.primary};
-    background: transparent;
-    border: none;
-
-    &:focus {
-        outline: none;
-    }
-
-    &::placeholder {
-        color: ${themeVars.layout.onSurface.secondary};
-    }
-
-    &:disabled {
-        cursor: not-allowed;
-        color: ${themeVars.layout.strokes};
-
-        &::placeholder {
-            color: ${themeVars.layout.strokes};
-        }
-    }
-`;
-
-const InnerIconContainer = styled.div<{ size: 'medium' | 'small' }>`
-    width: ${({ size }) => (size === 'small' ? '16px' : '20px')};
-    height: ${({ size }) => (size === 'small' ? '16px' : '20px')};
+const KeyframeContainer = styled.div`
     display: flex;
-    justify-content: center;
     align-items: center;
 `;
 
-const TrailingContainer = styled.div`
-    display: flex;
-    align-items: center;
-    gap: ${themeVars.spacing.s};
+const InputWrapper = styled.div`
+    flex: 1;
+    position: relative;
 `;
 
-const UnitsContainer = styled.div`
-    min-width: 24px;
+const HoverControlsContainer = styled.div`
+    position: absolute;
+    right: ${themeVars.spacing.s};
+    top: 50%;
+    transform: translateY(-50%);
     display: flex;
     align-items: center;
-    justify-content: center;
+    gap: ${themeVars.spacing.xs};
+    opacity: 0;
+    pointer-events: none;
+    transition: opacity ${themeVars.animation.duration.fast} ${themeVars.animation.timing.easeOut};
+
+    &[data-show='true'] {
+        opacity: 1;
+        pointer-events: auto;
+    }
```

Comment:
no hover controls should be in VisualEditorInput. It should work similar to VisualEditorSegmentedControl. Toggles should be after the Child input